routing/pathfind_test.go

```diff
@@ -718,20 +756,25 @@ func testBasicGraphPathFindingCase(t *testing.T, graphInstance *testGraphInstanc
 	target := graphInstance.aliasMap[test.target]
 	path, err := findPath(
 		nil, graphInstance.graph, nil, sourceNode, target,
-		ignoredVertexes, ignoredEdges, paymentAmt, test.feeLimit, nil,
+		ignoredVertexes, ignoredEdges, paymentAmt, test.pathRestrictions, nil,
 	)
-	if test.expectFailureNoPath {
-		if err == nil {
-			t.Fatal("expected no path to be found")
-		}
-		return
-	}
+
 	if err != nil {
-		t.Fatalf("unable to find path: %v", err)
+
+		if !test.expectError {
```

Comment:
Can just use `test.expectedFailure = nil` for cases where we don't expect an error? Then we won't need the `expectError` field.

Reply:
Would require creating a nill errortype. Should it be possible to throw a nill error?

Reply:
All methods return a `nil` error in the success case?

Reply:
But what do we put in `test.expectedFailure` if we remove `test.expectError`?
Currently we don't have a "nil" error.

Reply:
Can't you set `test.expectedFailure = nil` ?

Reply:
No, the expectedFailure is an int

Reply:
Can you make it `error` instead?

Reply:
If we did we’d still need to pass in the error code somewhere and get back to where we started